src/OVAL/probes/SEAP/sexp-parser.c

```diff
@@ -475,211 +504,199 @@ SEXP_t *SEXP_parse (const SEXP_psetup_t *psetup, char *buffer, size_t buflen, SE
                 assume_d (e_dsc.s_exp   != NULL, NULL);
                 assume_r (e_dsc.p_label != 128, NULL);
 
-                goto laddr(e_dsc.p_label);
+                dfa_state = d_states[e_dsc.p_label];
+        } else if (e_dsc.s_exp != NULL) {
+                dfa_state = S_NO_SEXP_ALLOC;
         }
 
-        if (e_dsc.s_exp != NULL)
-                goto L_NO_SEXP_ALLOC;
-
         for (;;) {
-                /*
-                 * Allocate an empty S-exp. The value or type will be assigned in the
-                 * subparser.
-                 */
-                assume_d (e_dsc.s_exp == NULL, NULL); /* no leaks */
-                e_dsc.s_exp = SEXP_new ();
-
-        L_NO_SEXP_ALLOC:
-                if (e_dsc.p_bufoff >= spb_len)
+                switch (dfa_state) {
+                case S_START:
+                        /*
+                         * Allocate an empty S-exp. The value or type will be assigned in the
+                         * subparser.
+                         */
+                        assume_d (e_dsc.s_exp == NULL, NULL); /* no leaks */
+                        e_dsc.s_exp = SEXP_new ();
+                        dfa_state = S_NO_SEXP_ALLOC;
```

Comment:
indents

Reply:
Unfortunately, this whole file was indented by 8 spaces. If we follow this indentation style, the result looks correct, even if the diff is weird. Do you think it would be better to change the indents in this file or in this function to tabs?

Reply:
Use tabs for indentation in new code. It's easier to just do it than fret about it too much IMO. Yes, sometimes that will be weird but over time we will converge the codebase to one code style.